src/components/smart_objects/src/smart_object.cc

```diff
@@ -734,16 +734,32 @@ void SmartObject::duplicate(const SmartObject& OtherObject) {
 void SmartObject::cleanup_data() {
   switch (m_type) {
     case SmartType_String:
-      delete m_data.str_value;
+      if (m_data.str_value) {
+        delete m_data.str_value;
+        m_data.str_value = nullptr;
+        m_type = SmartType_Null;
+      }
       break;
     case SmartType_Map:
-      delete m_data.map_value;
+      if (m_data.map_value) {
+        delete m_data.map_value;
+        m_data.map_value = nullptr;
+        m_type = SmartType_Null;
+      }
       break;
     case SmartType_Array:
-      delete m_data.array_value;
+      if (m_data.array_value) {
+        delete m_data.array_value;
+        m_data.array_value = nullptr;
+        m_type = SmartType_Null;
+      }
       break;
     case SmartType_Binary:
-      delete m_data.binary_value;
+      if (m_data.binary_value) {
+        delete m_data.binary_value;
+        m_data.array_value = nullptr;
```

Comment:
Copy paste error

Reply:
@JackLivio fixed in https://github.com/smartdevicelink/sdl_core/pull/3116/commits/0ecdff2aa896168e0e57f04bb31d48cd955ac67d